internal/query/projection/project_relational.go

```diff
@@ -0,0 +1,171 @@
+package projection
+
+import (
+	"context"
+
+	repoDomain "github.com/zitadel/zitadel/backend/v3/domain"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/eventstore/handler/v2"
+	"github.com/zitadel/zitadel/internal/repository/project"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+const (
+	ProjectRelationalTable                       = "zitadel.projects"
+	ProjectRelationalShouldAssertRoleCol         = "should_assert_role"
+	ProjectRelationalIsAuthorizationRequiredCol  = "is_authorization_required"
+	ProjectRelationalIsProjectAccessRequiredCol  = "is_project_access_required"
+	ProjectRelationalUsedLabelingSettingOwnerCol = "used_labeling_setting_owner"
+)
+
+type projectRelationalProjection struct{}
+
+func (*projectRelationalProjection) Name() string {
+	return ProjectRelationalTable
+}
+
+func newProjectRelationalProjection(ctx context.Context, config handler.Config) *handler.Handler {
+	return handler.NewHandler(ctx, &config, new(projectRelationalProjection))
+}
+
+func (p *projectRelationalProjection) Reducers() []handler.AggregateReducer {
+	return []handler.AggregateReducer{
+		{
+			Aggregate: project.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  project.ProjectAddedType,
+					Reduce: p.reduceProjectAdded,
+				},
+				{
+					Event:  project.ProjectChangedType,
+					Reduce: p.reduceProjectChanged,
+				},
+				{
+					Event:  project.ProjectDeactivatedType,
+					Reduce: p.reduceProjectDeactivated,
+				},
+				{
+					Event:  project.ProjectReactivatedType,
+					Reduce: p.reduceProjectReactivated,
+				},
+				{
+					Event:  project.ProjectRemovedType,
+					Reduce: p.reduceProjectRemoved,
+				},
+			},
+		},
+	}
+}
+
+func (p *projectRelationalProjection) reduceProjectAdded(event eventstore.Event) (*handler.Statement, error) {
+	e, ok := event.(*project.ProjectAddedEvent)
+	if !ok {
+		return nil, zerrors.ThrowInternalf(nil, "HANDL-Oox5e", "reduce.wrong.event.type %s", project.ProjectAddedType)
+	}
+
+	return handler.NewCreateStatement(
```

Comment:
We changed the way we implement the reduce functions by using the repositories instead of using the statements. See org domains for reference: https://github.com/zitadel/zitadel/blob/main/internal/query/projection/org_domain_relational.go

Reply:
It also gives you a first impression if the methods of the repository make sense.